api/src/task/notifications/closing_date_notification.py

```diff
@@ -0,0 +1,185 @@
+
+        # Find saved opportunities closing in two weeks that haven't been notified
+        stmt = (
+            select(UserSavedOpportunity)
+            .join(UserSavedOpportunity.opportunity)
+            .join(
+                OpportunitySummary, OpportunitySummary.opportunity_id == Opportunity.opportunity_id
+            )
+            .where(
+                # Check if closing date is within 24 hours of two weeks from now
+                and_(
+                    OpportunitySummary.close_date >= two_weeks_from_now - timedelta(hours=24),
+                    OpportunitySummary.close_date <= two_weeks_from_now + timedelta(hours=24),
+                ),
+                # Ensure we haven't already sent a closing reminder
+                ~exists().where(
+                    and_(
+                        UserOpportunityNotificationLog.user_id == UserSavedOpportunity.user_id,
+                        UserOpportunityNotificationLog.opportunity_id
+                        == UserSavedOpportunity.opportunity_id,
+                    )
+                ),
+            )
+        )
+
+        results = self.db_session.execute(stmt).scalars().all()
+        closing_date_opportunities: dict[UUID, list[UserSavedOpportunity]] = {}
+
+        for result in results:
+            user_id = result.user_id
+            closing_date_opportunities.setdefault(user_id, []).append(result)
+
+        logger.info(
+            "Collected closing date notifications",
+            extra={
+                "user_count": len(closing_date_opportunities),
+                "total_notifications": sum(
+                    len(closing_date_opportunities)
+                    for container in closing_date_opportunities.values()
+                ),
+            },
+        )
+        self.collected_data = closing_date_opportunities
+
+        return closing_date_opportunities
+
+    def prepare_notification(
+        self, closing_date_data: dict[UUID, list[UserSavedOpportunity]]
+    ) -> EmailData:
+
+        notification: dict[UUID, str] = {}
+        to_address_list: list[str] = []
+
+        for user_id, saved_items in closing_date_data.items():
+            user_email = self._get_user_email(user_id)
+            if not user_email:
+                continue
+            if len(saved_items) == 1:
+                # Single opportunity closing
+                opportunity = saved_items[0]
+                close_date_stmt = select(OpportunitySummary.close_date).where(
+                    OpportunitySummary.opportunity_id == opportunity.opportunity_id
+                )
+                close_date = self.db_session.execute(close_date_stmt).scalar_one_or_none()
+                if close_date is None:
+                    logger.warning(
+                        "No close date found for opportunity",
+                        extra={"opportunity_id": opportunity.opportunity_id},
+                    )
+                    continue
```

Comment:
This logic assumes an opportunity has a single opportunity summary which isn't correct, it can have two, and the logic of how that works is an entire task that sets the data up for you already. Don't write any queries, just do:
```py
close_date = opportunity.current_opportunity_summary.opportunity_summary.close_date if opportunity.current_opportunity_summary
```

We have relationships between these, don't write queries.